### PR TITLE
CFIN-304 display a message when no search results are returned

### PR DESCRIPTION
**Pull request by ce666-york**
As with #36 you can try this out in my AWS sandbox at https://y3cdpz4cgk.execute-api.eu-west-1.amazonaws.com/v1.

cfin-304

### CODE REVIEW

**Comment by ChristianBeddows** [comment]
I really like the sandbox preview for your code, a great way of visualising the changes.

**Review by ChristianBeddows — APPROVED** [review]

File: src/components/CourseSearchResults.js [hunk 1]
```diff
@@ -8,6 +8,10 @@ const CourseSearchResults = ({ isSuccessfulSearch, searchResults }) => {
         return <SearchFailedMessage />;
     }
 
+    if (!searchResults?.length) {
```
Comment (FergusMcGlynn) [inline src/components/CourseSearchResults.js:11]: Very minor point, but could this be simplified?
```suggestion
    if (!searchResults) {
```

**Review by FergusMcGlynn — APPROVED** [review]

File: src/components/CourseSearchResults.js [hunk 2]
```diff
@@ -31,4 +35,10 @@ const SearchFailedMessage = () => (
     </div>
 );
 
+const NoSearchResultsMessage = () => (
+    <div className="c-alert c-alert--info">
```
Comment (RichardAllum) [inline src/components/CourseSearchResults.js:39]: I think we could do this with a pattern Library `<Alert>` component with a className of info
```suggestion
    <Alert className="info">Sorry, your search did not return any results.</Alert>
```

Comment (ce666-york) [reply src/components/CourseSearchResults.js:39]: I didn't clock this, yes this feels like an improvement (although an `InfoAlert` would be even better)! As we have another alert already being used elsewhere, I'll look at refactoring them both at the same time in a separate commit

File: src/components/CourseSearchResults.js (hunk 1 above)
Comment (ce666-york) [reply src/components/CourseSearchResults.js]: Sounded good to me so I committed the suggestion, but it caused the tests to fail!

`!searchResults` just checks for the existence of the array I guess, and not the content. I've decided to make it `!searchResults || !searchResults.length` to be a bit more robust

Comment (FergusMcGlynn) [reply src/components/CourseSearchResults.js:11]: Oops... Sorry! :-/

**Review by RichardAllum — APPROVED** [review]

File: src/components/CourseSearchResults.js (hunk 1 above)
Comment (ce666-york) [reply src/components/CourseSearchResults.js:11]: No problem at all, it's a learning experience!

**Pull request merged**